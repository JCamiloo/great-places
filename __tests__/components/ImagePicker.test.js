import React from "react";
import ImagePicker from "../../components/ImagePicker";
import renderer from "react-test-renderer";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallow } from "enzyme";

Enzyme.configure({ adapter: new Adapter() });

describe("ImagePicker component tests", () => {
  let component;
  let button;

  test("ImagePicker component renders correctly", () => {
    const tree = renderer.create(<ImagePicker />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  beforeEach(() => {
    component = shallow(<ImagePicker />);
    button = component.find("Button");
    label = component.find("Text");
  });

  it("it renders default text", () => {
    const expectedText = "No image picked yet.";
    const text = label.children().text();
    expect(text).toEqual(expectedText);
  });

  // it("it renders button text", () => {
  //   const expectedText = "Take image";
  //   const text = button.text();
  //   expect(text).toEqual(expectedText);
  // });
});
