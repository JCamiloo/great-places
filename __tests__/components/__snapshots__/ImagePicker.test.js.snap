// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePicker component tests ImagePicker component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginBottom": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#ccc",
        "borderWidth": 1,
        "height": 200,
        "justifyContent": "center",
        "marginBottom": 10,
        "width": "100%",
      }
    }
  >
    <Text>
      No image picked yet.
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#fc9208",
            },
          ]
        }
      >
        Take image
      </Text>
    </View>
  </View>
</View>
`;
